### ISSUE TITLE
popgenWindows generating empty output

### ISSUE THREAD

**Issue by rafmarcondes** [simonhmartin/genomics_general#31]
Hi Simon,

The command below runs without error, and prints a lot of progress messages, but the output is an empty file. It creates the output file, but apparently nothing is written to it. What am I doing wrong? Thanks!

python /home/rmarco3/bin/simon-martin-scripts/popgenWindows.py --ploidy 2 -f phased -g asp_con_din_par_quality_filtered_ONLY.geno --windType coordinate -w 50000 -s 25000 --popsFile pops_file_for_martin -p din -p asp --verbose -o simon_stats_dinVasp

**Comment by rafmarcondes** [comment]
Correct. The output is empty. Not even a header. And adding -m 10 doesn't change it.

**Comment by rafmarcondes** [comment]
Is it supposed to write to the output file as it goes, or does it only write it at the end? If the latter, that might be the problem. It's a huge dataset, so I'm still waiting for a full run to complete. It's running for days and the output file is still empty though/

**Comment by simonhmartin** [comment]
Ah yes, with the buffering it might not write the output to disk until it is finished.
To speed it up, try multiple threads (-T).
